weto/src/main/java/fi/uta/cs/weto/util/NotificationManager.java

```diff
@@ -39,14 +37,225 @@ public void contextInitialized(ServletContextEvent event) {
+                                logger.debug("Something went wrong: " + e);
+                            }
+
+                            //Implement teacher notifications here
+                            if (triggeringPermissions.contains(permission.getType())) {
+                                int taskId = permission.getTaskId();
+                                Task task = Task.select1ById(courseCon, taskId);
+                                int rootTaskId = task.getRootTaskId();
+                                try {
+                                    int masterCourseId = CourseImplementation.select1ByDatabaseIdAndCourseTaskId(masterCon, databaseID, rootTaskId).getMasterTaskId();
+
+                                    // Create notification's message for teachers.
+                                    HashMap<String, String> templateValueMap = new HashMap<>();
+                                    templateValueMap.put("&permission;", WetoUtilities.getMessageResource(PermissionType.getType(permission.getType()).getProperty()));
+                                    templateValueMap.put("&task;", task.getName());
+                                    templateValueMap.put("&expirationTime;", new WetoTimeStamp(permission.getEndDate()).toString());
+
+                                    String notificationMessage = Notification.getMessageFromTemplate(Notification.PERMISSION_EXPIRATION, templateValueMap);
+
+                                    // Create link.
+                                    String link = WetoUtilities.getPackageResource("weto.appBaseUrl")
+                                            + "/viewPermissions.action?taskId=" + taskId
+                                            + "&tabId=" + Tab.PERMISSIONS.getValue()
+                                            + "&dbId=" + databaseID;
+
+
+                                    for (UserTaskView teacher : UserTaskView.selectByTaskIdAndClusterType(courseCon, rootTaskId, ClusterType.TEACHERS.getValue())) {
+                                        UserAccount courseAccount = UserAccount.select1ById(courseCon,teacher.getUserId());
+                                        UserAccount masterAccount = UserAccount.select1ByLoginName(masterCon,courseAccount.getLoginName());
+                                        Notification notification = new Notification(masterAccount.getId(), masterCourseId, Notification.PERMISSION_EXPIRATION, link);
+                                        notification.setMessage(notificationMessage);
+                                        notification.createNotification(masterCon, courseCon);
+
+                                    }
+                                }
+                                catch (Exception ignored) {
+                                }
+                            }
+
+                            // Check if permission is submission permission
+                            if (permission.getType() == 1) {

```

Comment:
Use the definitions in the PermissionType-class to compare against
```suggestion
                            if (permission.getType() == PermissionType.SUBMISSION.value) {
```